.github/workflows/smoke.yml

```diff
@@ -17,8 +17,12 @@ jobs:
       # e.g. if set 1 fails, continue with set 2 anyway
       fail-fast: false
     runs-on: ubuntu-latest
-    # Job named e.g. "Chrome stable, batch 1".
-    name: Chrome ${{ matrix.chrome-channel }}, batch ${{ matrix.smoke-test-shard }}
+    env:
+      # The total number of shards. Set dynamically when length of *single* matrix variable is
+      # computable. See https://github.community/t/get-length-of-strategy-matrix-or-get-all-matrix-options/18342
```

Comment:
Happy to change the wording here, as I assume it read like there is a solution (since the linked github discussion has an accepted solution), when really it was meant for background of the slightly different problem (multiple matrix dimensions) that isn't solved.